### ISSUE TITLE
failed: [429] Too Many Requests

### ISSUE THREAD

**Issue by sbrnag** [Microsoft/BotFramework-Emulator#70]
Hi,

   While testing my bot through BotFramework-Emulator i am getting the below error frequently.

Any idea how to resolve this. I am not doing too many requests just testing bot manually .

   Error: Request to 'https://7fec1104.ngrok.io/v3/botstate/emulator/conversations/kcn7g937c0jjde0c9/users/default-user' failed: [429] Too Many Requests
    at Request._callback (n:\ric\ricbotbybotframework\node_modules\botbuilder\lib\bots\ChatConnector.js:435:46)
    at Request.init.self.callback (n:\ric\ricbotbybotframework\node_modules\request\request.js:186:22)
    at emitTwo (events.js:87:13)

Thanks,
Nagnedra Prasad.

**Comment by eanders-MS** [comment]
Hi @sbrnag Due to a bug in the emulator, we're routing localhost requests through ngrok. A fix has been checked in and will be included in the next emulator release, due out next week. As a workaround, you can disable ngrok by clearing the path to ngrok from the emulator's App Settings dialog. Apologies for the inconvenience, and thank you for taking the time to report it.

**Comment by eanders-MS** [comment]
@sbrnag I'd assumed you were running your bot locally, but reading your post you don't make mention of it. Are running your bot locally or remotely?

**Comment by sbrnag** [comment]
Hi @eanders-MS Thanks for you reply. I am running bot locally. Will try by removing the ngrok path and continue.

**Comment by spghassenrjab** [comment]
I know this thread is closed, but I have a similar issue with a different scenario.
I am developing my bot using Node.js SDK in a vagrant box.
To test my bot using the emulator, at first I tried to forward the port 3978 from my host machine (Windows), where the emulator is installed to the guest machine (Ubuntu) where the bot is running.
The bot receives the messages but it can not reply as it will try to reply to localhost:53872 (The port that the emulator is listening to) which is only working on my host machine not the guest.
So I used ngrok to create the tunnel to localhost:3978 on my host machine (port 3978 still forwarded to my guest) and I used the generated URL to test my bot using the emulator.
The emulator can send and receive messages to and from the bot but sometimes I get this error
failed: [429] Too Many Requests
